### DISCUSSION TITLE
toData

### DISCUSSION THREAD

**Discussion by francoism90** [spatie/laravel-data#232] [Ideas]
Would it be possible to transform, but instead of an array, return a (Data) object?

Maybe I'm doing something wrong, but I now need to do this:
```php
$data = UserData::toArray();
echo $data['created_at']; // transformed timezone
```
It would be cool if you could use something like this:
```php
$data = UserData::toData(); // or toObject
echo  $data->created_at; // transformed timezone
```
Does this exists already?

Thanks!

**Comment by rubenvanassche** [comment]
What would be the use case here? You already have a data object so a syntax like `$data->created_at` is already possible.

By the way, these methods can only be called on an existing data object, not statically:

```php
UserData::toArray();
UserData::toData();
```

**Reply by francoism90** [reply]
@rubenvanassche In one project I'm using Livewire, which means `DataObject::toArray()` or `DataObject::toJson()` isn't very friendly.

```
$user = User::first()->getData();
$user->created_at // not transformed
```

```
$user = User::first()->getData()->toArray();
$user['created_at'] // transformed (e.g. with user timezone)
```

It would be cool if I could use the same syntax or provide my own `to..` method.

Something like this:
```
public function toObject(): object
    {
        return (object) $this->transform();
    }
```
Or with recursively:
```
public function toObject(): object
    {
        return json_decode(json_encode($this->transform()), false);
    }
```
I could write my own wrapper, but I was thinking if it would be possible to transform each value while keeping the dataobject.

**Reply by rubenvanassche** [reply]
Hmm, I get the point but I think it will get quite confusing what it would do since it transforms values. We try too keep in line with Laravel's `toArray` which transforms nested structures and data and `all` which doesn't do any transforming.

In this case I would opt for a custom trait adding the feature as you described above 😄

**Comment by francoism90** [comment]
@rubenvanassche Thanks again for your suggestion. I've used Laravel Fluent which converts it to exactly what I wanted. :)
